lib/image.js

```diff
@@ -0,0 +1,27 @@
+'use strict';
+const imagemin = require('imagemin');
+const imageminMozjpeg = require('imagemin-mozjpeg');
+const imageminPngquant = require('imagemin-pngquant');
+const gm = require('gm').subClass({ imageMagick: true });
+
+module.exports.compress = (file, level, callback) => {
+  const handleResult = result => {
+    if (result.length === 0) {
+      return callback('Unable to compress image');
+    }
+    callback(null, result[0].path);
+  };
+  const options = {
+    plugins: [imageminMozjpeg({ quality: level }), imageminPngquant({ quality: level })]
+  };
+  imagemin([file], 'build/', options).then(handleResult);
```

Comment:
That's what the 'mkdir' is doing there, making sure the 'build' directory is present where the files are going to be stored.